src/circuit/auto-relay.js

```diff
@@ -224,7 +233,32 @@ class AutoRelay {
       }
     }
 
-    // TODO: Try to find relays to hop on the network
+    // Try to find relays to hop on the network
+    try {
+      const cid = await namespaceToCid(RELAY_RENDEZVOUS_NS)
+      for await (const provider of this._libp2p.contentRouting.findProviders(cid)) {
+        if (!provider || !provider.id || !provider.multiaddrs || !provider.multiaddrs.length) {
```

Comment:
Are all these checks necessary? This feels like a bug in the API if so

Reply:
nope, I will simplify to check if `!provider || !provider.multiaddrs.length`.

Reply:
Provider should always exist, if it doesn't that's an issue. We really **should** have addresses too, otherwise the remote end is doing bad things, but that check is more reasonable.